pkg/operator/sync_common.go

```diff
@@ -28,145 +28,195 @@ func manageControllerNS(c serviceCAOperator) (bool, error) {
+	_, mod, err = resourceapply.ApplyRoleBinding(c.rbacv1Client, c.eventRecorder, requiredRoleBinding)
 	if err != nil {
-		return false, "rolebinding"
+		return err
 	}
+	*modified = *modified || mod
 
 	requiredSA := resourceread.ReadServiceAccountV1OrDie(v4_00_assets.MustAsset(resourcePath + "sa.yaml"))
-	_, saModified, err := resourceapply.ApplyServiceAccount(c.corev1Client, c.eventRecorder, requiredSA)
+	_, mod, err = resourceapply.ApplyServiceAccount(c.corev1Client, c.eventRecorder, requiredSA)
 	if err != nil {
-		return false, "serviceaccount"
+		return err
 	}
+	*modified = *modified || mod
 
-	return saModified, ""
+	return nil
 }
 
 // TODO manage rotation in addition to initial creation
-func manageSignerCA(client coreclientv1.SecretsGetter, eventRecorder events.Recorder) (*corev1.Secret, bool, error) {
+func manageSignerCA(client coreclientv1.SecretsGetter, eventRecorder events.Recorder, modified *bool) error {
 	secret := resourceread.ReadSecretV1OrDie(v4_00_assets.MustAsset("v4.0.0/service-serving-cert-signer-controller/signing-secret.yaml"))
-	existing, err := client.Secrets(secret.Namespace).Get(secret.Name, metav1.GetOptions{})
+	_, err := client.Secrets(secret.Namespace).Get(secret.Name, metav1.GetOptions{})
 	if !apierrors.IsNotFound(err) {
-		return existing, false, err
+		return err
 	}
+	name := serviceServingCertSignerName()
+	glog.V(4).Infof("generating signing CA: %s", name)
 
-	ca, err := crypto.MakeSelfSignedCAConfig(serviceServingCertSignerName(), 365)
+	ca, err := crypto.MakeSelfSignedCAConfig(name, 365)
 	if err != nil {
-		return existing, false, err
+		return err
 	}
 
 	certBytes := &bytes.Buffer{}
 	keyBytes := &bytes.Buffer{}
 	if err := ca.WriteCertConfig(certBytes, keyBytes); err != nil {
-		return existing, false, err
+		return err
 	}
 
 	secret.Data["tls.crt"] = certBytes.Bytes()
 	secret.Data["tls.key"] = keyBytes.Bytes()
+	_, mod, err := resourceapply.ApplySecret(client, eventRecorder, secret)
+	if err != nil {
+		return err
+	}
+	*modified = *modified || mod
 
-	return resourceapply.ApplySecret(client, eventRecorder, secret)
+	return nil
 }
 
-// TODO manage rotation in addition to initial creation
-func manageSignerCABundle(client coreclientv1.CoreV1Interface, eventRecorder events.Recorder) (*corev1.ConfigMap, bool, error) {
+func manageSignerCABundle(client coreclientv1.CoreV1Interface, eventRecorder events.Recorder, modified *bool) error {
 	configMap := resourceread.ReadConfigMapV1OrDie(v4_00_assets.MustAsset("v4.0.0/apiservice-cabundle-controller/signing-cabundle.yaml"))
-	existing, err := client.ConfigMaps(configMap.Namespace).Get(configMap.Name, metav1.GetOptions{})
+	_, err := client.ConfigMaps(configMap.Namespace).Get(configMap.Name, metav1.GetOptions{})
 	if !apierrors.IsNotFound(err) {
-		return existing, false, err
+		// Return if the configmap exists and we have not previously modified a resource (to force the update below).
+		if err == nil && !*modified {
+			return err
+		}
```

Comment:
```
err := everythings_on_fire
!typeNotFound(err) -> true
err == nil -> false

continuing with everythings_on_fire
```
Rename `modified` argument to `forceUpdate`, return new `modified` info separately from the function.